DatadogCore/Tests/Datadog/Core/Upload/DataUploadWorkerTests.swift

```diff
@@ -410,26 +421,26 @@ class DataUploadWorkerTests: XCTestCase {
             contextProvider: .mockAny(),
             uploadConditions: .alwaysUpload(),
             delay: DataUploadDelay(performance: UploadPerformanceMock.veryQuickInitialUpload),
-            featureName: .mockRandom()
+            featureName: .mockRandom(),
+            telemetry: telemetry
         )
 
         wait(for: [startUploadExpectation], timeout: 0.5)
         worker.cancelSynchronously()
 
         // Then
-        XCTAssertEqual(dd.telemetry.messages.count, 1)
+        XCTAssertEqual(telemetry.messages.count, 1)
 
-        guard case .error(_, let message, _, _) = dd.telemetry.messages.first else {
-            return XCTFail("An error should be send to `DD.telemetry`.")
+        guard case .error(_, let message, _, _) = telemetry.messages.first else {
+            return XCTFail("An error should be send to `telemetry`.")
```

Comment:
suggestion/ This gets quite repetitive in all tests. Recently I faced it when working on metrics telemetry. Solved by [adding helpers in `TU.TelemetryMocks`](https://github.com/DataDog/dd-sdk-ios/blob/d0d2b5448483768273421327ef0109e1b5728ce5/TestUtilities/Mocks/TelemetryMocks.swift#L83-L98):
```swift
_ = telemetry.messages.firstMetric(...)
_ = telemetry.messages.first.asMetric()
```
WDYT on facilitating other messages in similar or any other way? Otherwise it will be always a pain to add new telemetry assertion - `guard case` is quite inconvenient.